analyses/Linelist_cs_ebola_simulations.R

```diff
@@ -0,0 +1,278 @@
+
+# Script for individual-level data case study 4: disease with a delay from onset
+# to death shorter than onset to recovery (Ebola-like outbreak).
+# Methods to compare:
+# 1) Naive estimate of deaths/cases
+# 2) Retrospective cohort by omitting latest cases for which we don't expect to have an outcome (i.e., within 1 onset-death before real-time)
+# 3) Using {cfr} function estimate_outcomes to recalculate denominator
+# 4) Using {cfr} function cfr_static
+
+
+# Using simulist to simulate outbreak data
+# Columns that we need: onset date, column to indicate outcome, outcome date (recovery/death)
+
+library(data.table)
+library(zoo)
+library(tidyverse)
+library(simulist)
+library(epiparameter)
+library(incidence2)
+library(kableExtra)
+
+contact_distribution <- epidist(
+  disease = "ebola",
+  epi_dist = "contact distribution",
+  prob_distribution = "pois",
+  prob_distribution_params = c(mean = 3)
+)
+
+ebola_ip_meansd <- c(mean = 9.4, sd = 5.5)
+# From https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4560911/
+
+ebola_ip_params <- convert_summary_stats_to_params("lnorm", mean = ebola_ip_meansd[["mean"]], sd = ebola_ip_meansd[["sd"]])
+
+ip_ebola_epidist <- epidist(
+  disease = "ebola",
+  epi_dist = "infectious_period",
+  prob_distribution = "lnorm",
+  prob_distribution_params = c(meanlog = ebola_ip_params$meanlog, sdlog = ebola_ip_params$sdlog),
+  auto_calc_params = TRUE
+)
+
+# No distributions in epireview
+# summary statistics: mean = 18 and sd = 4
+
+onset_recovery_ebola <- convert_summary_stats_to_params("gamma", mean = 18, sd = 4)
+
+o_r_ebola_epidist <- epidist(
+  disease = "ebola",
+  epi_dist = "onset to recovery",
+  prob_distribution = "gamma",
+  prob_distribution_params = c(shape = onset_recovery_ebola[["shape"]], scale = onset_recovery_ebola[["scale"]])
+)
+
+o_d_ebola_epidist <- epidist_db(
+  disease = "ebola",
+  epi_dist = "onset to death",
+  single_epidist = TRUE
+)
+
+# Function rum_ebola_simulations uses simulist to create a list with n=100 (for now) linelists
+set.seed(487593478)
+simulations_ebola <- run_ebola_simulations(100)
```

Comment:
I think the `run_ebola_simulations()` function can be removed and replaced with an `lapply()`, which is good as it is less code to maintain as you run the analyses.

Reply:
Do you mean doing something like this:

`simulations_ebola <- lapply(1:100, function(x){
  sim_linelist(
    contact_distribution = contact_distribution,
    infectious_period = ip_ebola_epidist,
    onset_to_hosp = NULL,
    hosp_risk = NULL,
    hosp_death_risk = NULL,
    onset_to_death = o_d_ebola_epidist,
    onset_to_recovery = o_r_ebola_epidist,
    prob_infect = 0.7,
    outbreak_size = c(500, 2000),
    non_hosp_death_risk = 0.3
  )
})`

I've tried doing `lapply(1:100, sim_linelist, arguments)` but it wouldn't let me, should this work?